tests/rustdoc-ui/bad-intra-doc.rs

```diff
@@ -0,0 +1,13 @@
+#![no_std]
+#![deny(rustdoc::broken_intra_doc_links)]
+
+// regression test for https://github.com/rust-lang/rust/issues/54191
+
+/// this is not a link to [`example.com`] //~ERROR unresolved link
+///
+/// this link [`has spaces in it`].
+///
+/// attempted link to method: [`Foo.bar()`]
+///
+/// classic broken intra-doc link: [`Bar`]
```

Comment:
Sorry I didn't mention it, but these other lines also need error annotations on them.